# The Dynamic Impact of Foreign Direct Investment on Economic Growth in BRICS Countries


## Introduction
Foreign Direct Investment (FDI) is a crucial catalyst for global economic integration, particularly for emerging economies. The BRICS nations (Brazil, Russia, India, China, South Africa) have become significant FDI recipients, yet the precise impact of FDI on their economic growth remains a subject of debate, showing varied patterns across member states.

This research will rigorously investigate the dynamic and multifaceted relationship between FDI and economic growth in BRICS countries from 1995 to 2023. Employing advanced panel data methodologies, this study will explore FDI's long-run contribution to GDP, the mechanisms through which this occurs, and how critical macroeconomic factors (e.g., financial development, trade openness, gross fixed capital formation, and inflation) condition this relationship.

## Research Questions

**Main Research Question**  
What is the dynamic and conditional impact of Foreign Direct Investment on economic growth in BRICS countries, considering macroeconomic factors?

**Sub-Questions**
- What is the magnitude and persistence of FDI's influence on economic growth in BRICS nations?
- How do crucial macroeconomic factors (e.g., Gross Fixed Capital Formation, Trade, Inflation, Financial Development) affect the FDI-growth nexus?
- Has the relationship between FDI and economic growth in BRICS countries undergone significant changes following major global events, such as the 2008 Global Financial Crisis?


## Import packages

In [1]:
# Install required packages (Colab only)
!pip install linearmodels arch openpyxl --quiet

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from linearmodels.panel import PanelOLS, RandomEffects
from statsmodels.formula.api import ols
from arch.unitroot import ADF
from scipy import stats
from google.colab import files
from statsmodels.tsa.stattools import adfuller
from statsmodels.stats.outliers_influence import variance_inflation_factor





   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 18.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 985.3/985.3 kB 53.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 117.2/117.2 kB 11.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.9/43.9 kB 3.8 MB/s eta 0:00:00


## Upload and Read Data

In [18]:
# Read csv Data and convert it into a pandas DataFrame
df = pd.read_csv('BRICS_data.csv')
df = pd.DataFrame(df)

# Investigate shape of dataset, datatypes, column names
print("Shape of dataset:", df.shape)
print("\nData types:")
print(df.dtypes)
print("\nOriginal columns:")
print(df.columns.tolist())

Shape of dataset: (150, 9)

Data types:
Unnamed: 0                                             int64
Country Name                                          object
Year                                                   int64
Domestic credit to private sector (% of GDP)         float64
Foreign direct investment, net inflows (% of GDP)    float64
GDP per capita (constant 2015 US$)                   float64
Gross fixed capital formation (% of GDP)             float64
Inflation, consumer prices (annual %)                float64
Trade (% of GDP)                                     float64
dtype: object

Original columns:
['Unnamed: 0', 'Country Name', 'Year', 'Domestic credit to private sector (% of GDP)', 'Foreign direct investment, net inflows (% of GDP)', 'GDP per capita (constant 2015 US$)', 'Gross fixed capital formation (% of GDP)', 'Inflation, consumer prices (annual %)', 'Trade (% of GDP)']


## Preliminary Data Exploration and Data Cleaning


In [19]:
# Map long names to short ones used in the script:
rename_map = {
    'Country Name': 'country',
    'Year': 'year',
    'Foreign direct investment, net inflows (% of GDP)': 'fdi',
    'GDP per capita (constant 2015 US$)': 'gdp_pc',      # will be used to compute growth
    'Gross fixed capital formation (% of GDP)': 'gfcf',
    'Inflation, consumer prices (annual %)': 'inflation',
    'Trade (% of GDP)': 'trade',
    'Domestic credit to private sector (% of GDP)': 'domcred'
}

df = df.rename(columns=rename_map)

# Keep only relevant columns (if extra columns exist they are ignored)
keep_cols = ['country', 'year', 'gdp_pc', 'fdi', 'gfcf', 'inflation', 'trade', 'domcred']
df = df[[c for c in keep_cols if c in df.columns]]

# -------------------------
# 3) Type conversions & cleaning
# -------------------------
# Ensure year is integer and country is string
df['year'] = df['year'].astype(int)
df['country'] = df['country'].astype(str)

# Convert numeric-like columns to numeric (handles strings like "1,234" etc.)
num_cols = ['gdp_pc', 'fdi', 'gfcf', 'inflation', 'trade', 'domcred']
for col in num_cols:
    if col in df.columns:
        df[col] = pd.to_numeric(df[col], errors='coerce')

# Interpolate numeric missing values within each country (time series)
df = df.sort_values(['country', 'year'])
df[num_cols] = df.groupby('country')[num_cols].transform(lambda g: g.interpolate(method='linear', limit_direction='both'))

# Investigate first few rows
df.head(10)


country  year       gdp_pc       fdi       gfcf  inflation      trade  \
0  Brazil  1995  6596.335727  0.631586  20.286298  66.007034  16.984460   
1  Brazil  1996  6640.727007  1.475965  18.640654  15.757666  15.635591   
2  Brazil  1997  6764.858421  2.150453  19.122901   6.926713  16.576209   
3  Brazil  1998  6687.495236  3.340888  18.542349   3.195076  16.438585   
4  Brazil  1999  6621.636762  4.733770  17.016294   4.858447  20.982166   
5  Brazil  2000  6817.784241  5.033917  18.304488   7.044141  22.639761   
6  Brazil  2001  6823.033970  4.147594  18.418087   6.840359  26.936285   
7  Brazil  2002  6944.623375  3.253581  17.926251   8.450164  27.618357   
8  Brazil  2003  6941.440457  1.813401  16.604759  14.714920  28.140385   
9  Brazil  2004  7258.781852  2.713532  17.320233   6.597185  29.678252   

     domcred  
0  43.494525  
1  40.778494  
2  40.852237  
3  29.532261  
4  29.826910  
5  31.140986  
6  29.004038  
7  29.645142  
8  27.685670  
9  29.372770

# Calculating GDP per capita growth from GDP per capita

In [21]:
# Compute annual percent change of GDP per capita (multiply by 100 for percent)
df['gdp_growth'] = df.groupby('country')['gdp_pc'].transform(lambda x: x.pct_change() * 100)

# Verify the dataset

In [29]:
# Filtering the dataset to include data from 2000 to 2023
# df_final = df[(df['year'] >= 2000) & (df['year'] <= 2023)]

# print("\nAfter cleaning, sample rows:")
# display(df.head())

# print("\nData types:")
# print(df.dtypes)

# missing_rows = df[df['gdp_growth'].isna()].shape[0]
# print(f"\nNumber of missing rows: {missing_rows}")

# df.to_csv('df_growth.csv', index=False)
# files.download('df_growth.csv')

df_final = pd.read_csv('df_growth.csv')
df_final.head()

country  year       gdp_pc       fdi       gfcf  inflation      trade  \
0  Brazil  1995  6596.335727  0.631586  20.286298  66.007034  16.984460   
1  Brazil  1996  6640.727007  1.475965  18.640654  15.757666  15.635591   
2  Brazil  1997  6764.858421  2.150453  19.122901   6.926713  16.576209   
3  Brazil  1998  6687.495236  3.340888  18.542348   3.195076  16.438585   
4  Brazil  1999  6621.636762  4.733770  17.016294   4.858447  20.982166   

     domcred  gdp_growth  
0  43.494525    2.632056  
1  40.778494    0.672969  
2  40.852237    1.869244  
3  29.532261   -1.143604  
4  29.826910   -0.984800

# Descriptive Statistics

In [30]:

# -------------------------
# 6) Descriptive stats & correlation (numeric only)
# -------------------------
numeric_df = df_final.select_dtypes(include=[np.number])
print("\nDescriptive statistics (numeric variables):")
display(numeric_df.describe())




Descriptive statistics (numeric variables):


year        gdp_pc         fdi        gfcf   inflation  \
count   150.000000    150.000000  150.000000  150.000000  150.000000   
mean   2009.500000   5803.348408    2.138158   24.640750    8.706439   
std       8.684438   3162.951223    1.499088    9.034801   18.329569   
min    1995.000000    620.699954   -1.736810   13.051369   -1.401473   
25%    2002.000000   2672.306541    0.947852   17.766650    3.730373   
50%    2009.500000   6021.274678    1.889431   20.953588    5.679009   
75%    2017.000000   8637.367960    3.237892   31.284409    8.340240   
max    2024.000000  13121.676990    9.660265   44.075543  197.414268   

            trade     domcred  gdp_growth  
count  150.000000  150.000000  150.000000  
mean    42.572733   75.277973    3.537981  
std     12.610015   43.869472    3.949630  
min     15.635591   16.837772   -7.827749  
25%     33.638121   41.983966    1.145551  
50%     44.545429   57.015257    3.633641  
75%     51.674817  112.195288    6.347265  
max     69.393282  194.165997   13.555366

# Descriptive Statistics – Summary Interpretation

**Sample size:** 150 observations across BRICS countries (Brazil, Russia, India, China, South Africa) from **1995–2024** (balanced panel).

## Variable Insights

### GDP per capita (`gdp_pc`)
- **Mean:** ≈ $5,803  
- **Variation:** Large (std ≈ $3,163)  
- **Range:** $621 (low-income) to $13,122 (upper-middle income)

### Foreign Direct Investment (`fdi`) (% of GDP)
- **Mean:** ≈ 2.14%  
- **Range:** Negative inflows (disinvestment) to high inflows (max ≈ 9.66%)

### Gross Fixed Capital Formation (`gfcf`) (% of GDP)
- **Mean:** ≈ 24.64%  
- **Dispersion:** Moderate — relatively stable investment rates across time

### Inflation (annual %)
- **Mean:** ≈ 8.71%  
- **Volatility:** Very high — max ≈ 197% (indicative of crisis episodes)

### Trade Openness (`trade`) (% of GDP)
- **Mean:** ≈ 42.57%  
- **Range:** 15.64% to 69.39% — reflecting varied global integration levels

### Domestic Credit to Private Sector (`domcred`) (% of GDP)
- **Mean:** ≈ 75.28%  
- **Dispersion:** Very high (std ≈ 43.87%) — indicates differences in financial sector depth

### GDP Growth (`gdp_growth`) (annual %)
- **Mean:** ≈ 3.54%  
- **Range:** -7.83% (deep recessions) to 13.56% (high growth surges)

---

## Key Takeaways
- Substantial heterogeneity across BRICS in **economic size**, **openness**, and **macroeconomic stability**.  
- **Inflation** and **domestic credit** show the **highest dispersion**, reflecting structural differences in monetary and financial systems.  
- **FDI flows** are modest on average but display **notable volatility**.


## Correlation Matrix

In [31]:
# print("\nCorrelation matrix (numeric variables):")
# plt.figure(figsize=(8,6))
# sns.heatmap(numeric_df.corr(), annot=True, fmt=".2f", cmap='coolwarm', linewidths=0.5)
# plt.title("Correlation matrix (numeric variables)")
# plt.show()
numeric_df.corr()

year    gdp_pc       fdi      gfcf  inflation     trade  \
year        1.000000  0.473305 -0.084974  0.106739  -0.280260  0.179491   
gdp_pc      0.473305  1.000000  0.034330 -0.191755  -0.046354 -0.082300   
fdi        -0.084974  0.034330  1.000000  0.082418  -0.141136 -0.089775   
gfcf        0.106739 -0.191755  0.082418  1.000000  -0.165321 -0.001185   
inflation  -0.280260 -0.046354 -0.141136 -0.165321   1.000000  0.135893   
trade       0.179491 -0.082300 -0.089775 -0.001185   0.135893  1.000000   
domcred     0.300380  0.252138  0.012761  0.393694  -0.292650  0.205075   
gdp_growth -0.133902 -0.257076  0.219410  0.590059  -0.183706  0.144297   

             domcred  gdp_growth  
year        0.300380   -0.133902  
gdp_pc      0.252138   -0.257076  
fdi         0.012761    0.219410  
gfcf        0.393694    0.590059  
inflation  -0.292650   -0.183706  
trade       0.205075    0.144297  
domcred     1.000000    0.131866  
gdp_growth  0.131866    1.000000

# Correlation Analysis – Summary Interpretation

This section provides an overview of the pairwise correlations among key macroeconomic variables for BRICS countries (1995–2024).

## Key Observations

- **GDP per capita (`gdp_pc`)**  
  - Positively correlated with **gfcf** (0.47) and **domcred** (0.25), indicating that higher-income countries tend to have more investment and financial sector development.  
  - Negatively correlated with **gdp_growth** (-0.26), suggesting higher-income countries experienced relatively lower growth rates in this sample.

- **Foreign Direct Investment (`fdi`)**  
  - Weak correlations with most variables, slightly positive with **gdp_growth** (0.22), implying modest growth benefits from FDI.  
  - Slight negative association with **inflation** (-0.14).

- **Gross Fixed Capital Formation (`gfcf`)**  
  - Strongly positively correlated with **gdp_growth** (0.59), indicating investment drives growth.  
  - Moderate positive correlation with **domcred** (0.39), reflecting the role of finance in supporting investment.

- **Inflation**  
  - Negatively correlated with **gdp_pc** (-0.05) and **gdp_growth** (-0.18), suggesting that higher inflation may be associated with slower growth and lower per capita income.  
  - Slight positive correlation with **domcred** (0.14), reflecting mixed financial conditions.

- **Trade Openness (`trade`)**  
  - Weak correlations with most variables; slightly positive with **domcred** (0.21) and **gdp_growth** (0.14), indicating trade may modestly support growth and financial development.

- **Domestic Credit (`domcred`)**  
  - Positively correlated with **gfcf** (0.39) and **gdp_pc** (0.25), suggesting financial depth supports investment and income levels.  
  - Weak positive correlation with **gdp_growth** (0.13).

- **GDP Growth (`gdp_growth`)**  
  - Strongest positive association with **gfcf** (0.59), highlighting the key role of investment in growth.  
  - Negative correlation with **gdp_pc** (-0.26), consistent with convergence patterns (poorer countries grow faster).

## Overall Insights
- **Investment (`gfcf`)** is the most strongly growth-linked variable in the panel.  
- **Financial development (`domcred`)** supports both investment and income but has a weaker link to growth directly.  
- **FDI and trade** show only modest associations with other macroeconomic indicators.  
- The results suggest **heterogeneous dynamics** across BRICS countries, with investment and domestic finance as key drivers of economic performance.


In [32]:
# 4. VIF (Multicollinearity Check)
# ================================
X = df_final[['fdi', 'gfcf', 'inflation', 'trade', 'domcred', 'gdp_pc']]
vif_data = pd.DataFrame()
vif_data['Variable'] = X.columns
vif_data['VIF'] = [variance_inflation_factor(X.values, i) for i in range(X.shape[1])]
print(vif_data)

    Variable       VIF
0        fdi  2.943445
1       gfcf  7.587014
2  inflation  1.420499
3      trade  7.903618
4    domcred  6.189195
5     gdp_pc  4.014330


# Multicollinearity Check – VIF Analysis

Variance Inflation Factor (VIF) was calculated for all explanatory variables to assess multicollinearity.

## VIF Results

- **FDI (`fdi`)**: 2.94  
  - Low VIF, suggesting minimal multicollinearity with other variables.

- **Gross Fixed Capital Formation (`gfcf`)**: 7.59  
  - Moderate to high VIF, indicating some multicollinearity concerns.

- **Inflation (`inflation`)**: 1.42  
  - Very low VIF, no multicollinearity concerns.

- **Trade Openness (`trade`)**: 7.90  
  - Moderate to high VIF, some correlation with other predictors.

- **Domestic Credit (`domcred`)**: 6.19  
  - Moderate multicollinearity present.

- **GDP per Capita (`gdp_pc`)**: 4.01  
  - Acceptable VIF, some correlation with other variables but not severe.

## Interpretation

- VIF values above 5–10 are typically considered indicative of multicollinearity.  
- Here, **gfcf**, **trade**, and **domcred** show moderate multicollinearity, which should be monitored in regression analysis.  
- **FDI**, **inflation**, and **gdp_pc** have low to moderate VIF, suggesting reliable contribution to the model without severe collinearity issues.


## Conclusion and Next Steps

This notebook has successfully initiated an investigation into the dynamic impact of Foreign Direct Investment on economic growth in BRICS countries from 1995 to 2024.

We have completed the crucial initial stages of data handling:
- Data loading and preliminary inspection.
- Cleaning and renaming of variables for clarity.
- Calculation of GDP per capita growth, the key dependent variable.
- Basic descriptive statistics and correlation analysis, providing initial insights into the relationships between variables and highlighting the heterogeneity within the BRICS group.
- Multicollinearity checks using VIF, confirming that while some variables show moderate correlation, it is not at a level that would severely impede regression analysis at this stage.

The next critical step in preparing the data for rigorous panel data analysis is to formally test for stationarity and apply necessary transformations.

**Next Steps: Stationarity Tests and Transformation**

Time series data often exhibit non-stationarity, which can lead to spurious regression results. We will perform Augmented Dickey-Fuller (ADF) tests for each variable and each country to determine if the series are stationary.

Based on the ADF test results:
- Variables found to be stationary will be used directly in the regression models.
- Variables found to be non-stationary will be differenced (taking the first difference) to achieve stationarity.

After performing these tests and transformations, we will proceed to the core econometric modeling using panel data techniques (e.g., Fixed Effects, Random Effects) to address the research questions outlined in the introduction. This will allow us to estimate the impact of FDI and other macroeconomic factors on GDP growth in the BRICS nations.